opentech/apply/review/forms.py

```diff
@@ -78,45 +79,29 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def validate_unique(self):
-        super().validate_unique()
-
+        # update the instance data prior to validating uniqueness
         self.instance.submission = self.submission
         self.instance.author = self.request.user
         self.instance.review = self.cleaned_data
 
-        try:
-            self.instance.validate_unique()
-        except ValidationError as e:
-            self._update_errors(e)
+        super().validate_unique()
 
     def save(self, commit=True):
+        self.instance.score = self.calculate_score()
 
-        score_fields = self.get_score_fields()
+        self.instance.recommendation = self.cleaned_data['recommendation']
 
-        if score_fields:
-            items = 0
-            total = 0
-
-            for field in score_fields:
-                try:
-                    value = int(self.cleaned_data[field])
-                    if value < 90:
-                        items = items + 1
-                        total = total + value
-                except KeyError:
-                    pass
+        super().save()
 
-            try:
-                self.instance.score = total / items
-            except ZeroDivisionError:
-                pass
+    def caclulate_score(self):
+        scores = list()
 
-        try:
-            self.instance.recommendation = self.cleaned_data['recommendation']
-        except KeyError:
-            pass
+        for field in self.get_score_fields():
+            score = int(self.cleaned_data.get(field, NA))
+            if score != NA:
+                scores.append(score)
 
-        super().save()
+        return sum(scores) / len(scores)
```

Comment:
so what happens when there are no score fields, thus `len(scores)` is 0?

Reply:
I was missed it out. good spot.